email_totals/handler.py

```diff
@@ -0,0 +1,315 @@
+
+    # if this environment variable is set, only send emails to recipients
+    # listed as approved recipients, otherwise send all valid emails
+    if os.environ['RESTRICT'] == 'True':
+        restrict_send = True
+        print(f'Only sending emails to: {restrict_to}')
+    else:
+        restrict_send = False
+
+    # loop over all our totals
+    for k, v in all_totals.items():
+        if v < min_value:
+            print(f"Skipping entry less than ${min_value}: {k} (${v})")
+
+        elif restrict_send:
+            if k in restrict_to:
+                report_dict[k] = v
+            else:
+                print(f"Restricting email: {k} (${v})")
+
+        else:
+            if k.endswith(sagebase_email) or k.endswith(sagebio_email):
+                report_dict[k] = v
+
+            elif k.endswith(synapse_email):
+                if k in team_sage:
+                    report_dict[k] = v
+                else:
+                    print(f"Skipping external synapse user: {k} (${v})")
+
+            else:
+                print(f"Skipping invalid email: {k} (${v})")
+
+    return report_dict
+
+
+#--------------------------------------------------------------------------------------------------
+# Compile HTML for E-mail Body
+
+def create_and_send_emails(target_totals, compare_totals):
+    '''
+    Build an HTML email from a string template and give it to our SES client
+    '''
+
+    html_title = '<h2>AWS Monthly Cost Report Summary</h2><br/>'
+
+    def _build_email(target_total, previous_total):
+        '''
+        The actual string template for the email
+        '''
+
+        if previous_total:
+            pct_change = (target_total / previous_total) - 1
+        else:
+            pct_change = 1
+
+        return ('You are receiving this email because you have been tagged as a '
+                f'resource owner in AWS. Your tagged resources total ${target_total:.2f} '
+                f'for the past month ({pct_change:.2%} change from the month prior). '
+                'You can use AWS Cost Explorer to analyze those expenses: '
+                '<a href="https://sagebionetworks.jira.com/wiki/spaces/IT/pages/2756935685/Using+AWS+Cost+Explorer">'
+                'Using AWS Cost Explorer</a>')
+
+
+    for email, total in target_totals.items():
+        print(f'Processing email for {email} (${total})')
+
+        compare_total = None
+        if email in compare_totals:
+            compare_total = compare_totals[email]
+
+        email_body = html_title + _build_email(total, compare_total)
+
+        send_report_email(email, email_body)
+
+
+#--------------------------------------------------------------------------------------------------
+# Compile and send HTML E-mail
+
+def send_report_email(recipient, body_html):
+
+    sender = os.environ['SENDER']
+    subject = "AWS Monthly Cost Report for Selected Accounts"
+
+    # The email body for recipients with non-HTML email clients.
+    body_text = ("Amazon SES\r\n"
```

Comment:
should the content of this be the same as the passed in body_html?

Reply:
no, body_text is meant to be explicitly non-html. and i tried to preserve the send_report_email function from the lambda in the AWS blog post as much as possible. but i suppose i've removed most of the html from the email, so it wouldn't be hard to generate a body_text that replaces the last bit of html with plain text